ecosystem/python/sdk/aptos_sdk/transactions.py

```diff
@@ -0,0 +1,453 @@
+
+    def natural(
+        module: str,
+        function: str,
+        ty_args: List[TypeTag],
+        args: List[TransactionArgument],
+    ) -> ScriptFunction:
+        module_id = ModuleId.from_str(module)
+
+        byte_args = []
+        for arg in args:
+            byte_args.append(arg.encode())
+        return ScriptFunction(module_id, function, ty_args, byte_args)
+
+    def deserialize(deserializer: Deserializer) -> ScriptFunction:
+        module = ModuleId.deserialize(deserializer)
+        function = deserializer.str()
+        ty_args = deserializer.sequence(TypeTag.deserialize)
+        args = deserializer.sequence(Deserializer.bytes)
+        return ScriptFunction(module, function, ty_args, args)
+
+    def serialize(self, serializer: Serializer):
+        self.module.serialize(serializer)
+        serializer.str(self.function)
+        serializer.sequence(self.ty_args, Serializer.struct)
+        serializer.sequence(self.args, Serializer.bytes)
+
+
+class WriteSet:
+    def __init__(self):
+        raise NotImplementedError
+
+    def deserialize(deserializer: Deserializer) -> WriteSet:
+        raise NotImplementedError
+
+    def serialize(self, serializer: Serializer):
+        raise NotImplementedError
+
+
+class ModuleId:
+    address: AccountAddress
+    name: str
+
+    def __init__(self, address: AccountAddress, name: str):
+        self.address = address
+        self.name = name
+
+    def __eq__(self, other: ModuleId) -> bool:
+        return self.address == other.address and self.name == other.name
+
+    def __str__(self) -> str:
+        return f"{self.address}::{self.name}"
+
+    def from_str(module_id: str) -> ModuleId:
+        split = module_id.split("::")
+        return ModuleId(AccountAddress.from_hex(split[0]), split[1])
+
+    def deserialize(deserializer: Deserializer) -> ModuleId:
+        addr = AccountAddress.deserialize(deserializer)
+        name = deserializer.str()
+        return ModuleId(addr, name)
+
+    def serialize(self, serializer: Serializer):
+        self.address.serialize(serializer)
+        serializer.str(self.name)
+
+
+class TransactionArgument:
+    value: typing.Any
+    encoder: typing.Callable[[Serializer, typing.Any], bytes]
+
+    def __init__(
+        self,
+        value: typing.Any,
+        encoder: typing.Callable[[Serializer, typing.Any], bytes],
+    ):
+        self.value = value
+        self.encoder = encoder
+
+    def encode(self) -> bytes:
+        ser = Serializer()
+        self.encoder(ser, self.value)
+        return ser.output()
+
+
+class SignedTransaction:
+    transaction: RawTransaction
+    authenticator: Authenticator
+
+    def __init__(self, transaction: RawTransaction, authenticator: Authenticator):
+        self.transaction = transaction
+        self.authenticator = authenticator
+
+    def __str__(self) -> str:
+        return f"Transaction: {self.transaction}Authenticator: {self.authenticator}"
```

Comment:
Put a space between transaction and authenticator

Reply:
from a code perspective yes, but that would change the output, so this is actually the preferred approach :/